docs/navigation-key.md

```diff
@@ -0,0 +1,33 @@
+---
+id: navigation-key
+title: Using the navigation key
+sidebar_label: Using the navigation key
+---
+
+The `key` parameter comes up repeatedly across different navigation functions. Let's take a look at a summary of its use cases:
+
+### Usage with the [`navigate`](/docs/navigation-actions.html#navigate) call
+
+TL;DR: The behavior of `navigate` together with `key` depends on the version of react-navigation you use. In v1, key has to be provided for navigate to behave idempotently. In v2, this is no longer needed.
```

Comment:
I think we should only explain the behavior with the latest version of react-navigation in these docs. We can add a note that links to more information about how it worked in v1, eg:

> Note: the behavior of `navigate` without a `key` is significantly different in the 1.x series of releases. Read more (link to gist or something)